src/hotspot/hotspots.ts

```diff
@@ -369,3 +370,39 @@ export function showHotspotDetails(hotspotDetails: ShowRuleDescriptionParams, ho
 
   hotspotDetailsPanel.reveal();
 }
+
+export async function changeHotspotStatus(hotspotServerKey: string, fileUriAsSting: string,
+                                                  languageClient: SonarLintExtendedLanguageClient) {
+  const fileUri = vscode.Uri.parse(fileUriAsSting);
+  const workspaceFolder = vscode.workspace.getWorkspaceFolder(fileUri);
+  return doChangeHotspotStatus(hotspotServerKey, fileUriAsSting, workspaceFolder, languageClient);
+}
+
+// visible for testing
+export async function doChangeHotspotStatus(hotspotServerKey: string, fileUriAsSting: string,
+                                            workspaceFolder:vscode.WorkspaceFolder,
+                                            languageClient: SonarLintExtendedLanguageClient) {
+  const allowedHotspotStatuses =
+    await languageClient.getAllowedHotspotStatuses(hotspotServerKey, workspaceFolder.uri.toString(), fileUriAsSting);
+  if (allowedHotspotStatuses == null) {
+    return Promise.resolve();
+  }
+  if (!allowedHotspotStatuses.permitted) {
+    vscode.window.showWarningMessage(
+      `Not permitted to change hotspot status. Reason: ${allowedHotspotStatuses.notPermittedReason}`);
+    return Promise.resolve();
+  }
+  if (allowedHotspotStatuses.allowedStatuses.length === 0) {
+    vscode.window.showInformationMessage(
+      `There are no allowed statuses to set for this hotspot`);
+    return Promise.resolve();
+  }
+  const statusQuickPickItems = allowedHotspotStatuses.allowedStatuses;
+  const chosenStatus = await vscode.window.showQuickPick(statusQuickPickItems);
```

Comment:
As discussed, this QuickPick could also use a placeholder "prompt" text, like the issues one